pinot-common/src/main/java/org/apache/pinot/common/datablock/MetadataBlock.java

```diff
@@ -36,108 +39,129 @@
  */
 public class MetadataBlock extends BaseDataBlock {
 
-  private static final ObjectMapper JSON = new ObjectMapper();
-
+  private static final Logger LOGGER = LoggerFactory.getLogger(MetadataBlock.class);
   @VisibleForTesting
-  static final int VERSION = 1;
+  static final int VERSION = 2;
```

Comment:
Are we going to use different version for different block types (data block is still with v1)? I feel it will be safer to have all data blocks in the same version. Since we are not going to really maintain 2 versions and add flag to switch, should we just keep it as v1?

Reply:
I think it is cleaner to read a different version and explicitly write in the code what we do when a block of an older version is received. Instead of crossing our fingers knowing that v1 uses varsize while v2 uses metadata, now in `MetadataBlock.deserialize` we an explicit code saying what we are doing.

I'm open to increase the version of the other blocks as well.

Reply:
We use version to maintain backward compatible. When introducing a new version, we first add the support for both versions, then after all nodes are upgraded, use a flag to switch the versions. After it is stable, remove the flag and old version support.
Since we are not following the above process, version is not really checked when deserializing. I don't have strong opinion on whether to bump it. Just pointing it out that it is not necessary.

Reply:
I think we are not going to add support for 2nd version in the code we have in master, but the code in this PR actually reads 1st and 2nd version. We don't need to do the former because we are lucky and the binary format should be compatible (though we may need to verify that adding a test in master).

The implementation in both branches mostly ignore metrics in the other, but even they ignore the content, they are able to read them. We are just ignoring the content because we think it is not worth to actually convert from one format to the other given the impact will be to do not have all stats during the upgrade, but we could do that if needed.

I think it is fine to increase the version.